### DISCUSSION TITLE
KyuubiHistoryServerPlugin usage

### DISCUSSION THREAD

**Discussion by hanna-liashchuk** [apache/kyuubi#5267] [Q&A] [answered]
Hi, I found an interesting class here - KyuubiHistoryServerPlugin
But I cannot find anywhere how to use it. Any advice?

**Comment by cxzl25** [comment]
https://github.com/apache/kyuubi/blob/2371cc62f3757e643766865f95f5a6f2b46b7cf8/externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/spark/ui/KyuubiHistoryServerPlugin.scala#L30-L33

**Comment by hanna-liashchuk** [comment]
@cxzl25 
Ok, so I added the jar to /opt/spark/jars and set 
```
spark.kyuubi.engine.ui.stop.enabled     true
spark.ui.killEnabled                    true
```
as my spark conf

I can see the tab "Kyuubi Query Engine" but I cannot see the "kill" button
I found this commit https://github.com/apache/kyuubi/issues/3220

Could you please explain it? And what should I do to be able to kill spark engines in the UI?

Thanks!

**Reply by cxzl25** [reply, marked as answer]
The history server displays both completed and incomplete Spark jobs. 
But it does not support the kill operation, so there is no need to display the kill operation in the history server.
You can find `Tracking URL` or `application web UI` in the logs.
